CadbaseLibrary/CadbaseLibrary.FCMacro

```diff
@@ -0,0 +1,837 @@
+                # else:
+                #     FreeCAD.Console.PrintWarning(new_dir)
+                #     FreeCAD.Console.PrintWarning(' : is a directory\n')
+
+                new_info_file = new_dir + "/component"
+
+                if os.path.exists(new_info_file):
+                    if os.path.isdir(new_info_file):
+                        os.rmdir(new_info_file)
+                    else:
+                        os.remove(new_info_file)
+
+                f = open(new_info_file, "x")
+                data = json.dumps(component, default=lambda o: o.__dict__, indent=4)
+                f.write(data)
+                f.close()
+                # FreeCAD.Console.PrintWarning(new_info_file + '\n')
+
+        # data_file = open(new_info_file, 'r')
+        # component_data = json.loads(data_file.read(), object_hook=lambda d: SimpleNamespace(**d))
+        # FreeCAD.Console.PrintWarning(component_data.uuid)
+        # data_file.close()
+    FreeCAD.Console.PrintWarning('Finished\n')
+
+def updateComponent(LIBRARYPATH, component_path):
+
+    # FreeCAD.Console.PrintWarning('updateComponent start\n')
+    data = parsingGpl(LIBRARYPATH)
+    if isinstance(data, SimpleNamespace):
+
+        if len(data.componentModifications) == 0:
+
+            FreeCAD.Console.PrintWarning('No modifications for the component\n')
+
+        else:
+
+            for modification in data.componentModifications:
+                new_dir = component_path + "/" + modification.modificationName
+                # FreeCAD.Console.PrintWarning(new_dir)
+
+                if not os.path.exists(new_dir):
+                    os.mkdir(new_dir)
+                # else:
+                #     FreeCAD.Console.PrintWarning(new_dir)
+                #     FreeCAD.Console.PrintWarning(' : exists\n')
+
+                if not os.path.isdir(new_dir):
+                    os.remove(new_dir)
+                    os.mkdir(new_dir)
+                # else:
+                #     FreeCAD.Console.PrintWarning(new_dir)
+                #     FreeCAD.Console.PrintWarning(' : is a directory\n')
+
+                new_info_file = new_dir + "/modification"
+
+                if os.path.exists(new_info_file):
+                    if os.path.isdir(new_info_file):
+                        os.rmdir(new_info_file)
+                    else:
+                        os.remove(new_info_file)
+
+                f = open(new_info_file, "x")
+                data = json.dumps(modification, default=lambda o: o.__dict__, indent=4)
+                f.write(data)
+                f.close()
+                # FreeCAD.Console.PrintWarning(new_info_file + '\n')
+    FreeCAD.Console.PrintWarning('Finished\n')
+
+def updateComponentModificaion(LIBRARYPATH, modification_path, modification_uuid):
+
+    # FreeCAD.Console.PrintWarning('Getting files\n')
+
+    data = parsingGpl(LIBRARYPATH)
+    if isinstance(data, SimpleNamespace):
+
+        if len(data.componentModificationFilesets) == 0:
```

Comment:
Would `if data.componentModificationFilesets:` work?

Reply:
No, sorry, but this doesn't work if we have an array with no elements. I mean, in this case, an empty array and an array with elements will return `true` and then we need add other condition for check for case empty array.
But I'm not good at python, so I might be wrong and there is a solution that will be better and faster than this. Please tell me if you know a better solution)

Reply:
> an empty array [...] will return `true`

This is not correct, an empty list cast as bool as False:
```python
>>> bool([])
False
>>> bool([False])
True
```
Do you get my point?

Reply:
Oh you're right. Sorry, I don't even know why it didn't work for me when I tried it before.
I did it now and it works. Thank you for the explanation despite my stupidity!